packages/nimble-components/src/table/specs/table-data-api.md

```diff
@@ -0,0 +1,130 @@
+# Table Data API
+
+## Problem Statement
+
+Table/Data-grid components can have a variety of ways to introduce data into it. We should make it clear how we expect users to introduce data into the `nimble-table` and address the implications that has on the API.
+
+## Links To Relevant Work Items and Reference Material
+
+[FAST data-grid spec](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-foundation/src/data-grid/data-grid.spec.md) (see `rowsData` in _properties:_ section)
+
+[TanStack Table API](https://tanstack.com/table/v8/docs/api/core/table)
+
+## Implementation / Design
+
+To use the TanStack Table library it expects its `data` to be provided as an array of some arbitrary type. Much of its API is generic to the type of the object that represents a single row of data.
+
+The `FASTDataGrid` has similar expectations for its [data API](https://github.com/microsoft/fast/blob/416dc9167e9d41e6ffe11d87ed79b2f455357923/packages/web-components/fast-foundation/src/data-grid/data-grid.ts#L193), provided via a `rowsData` property of type `object[]`. Columns are determined by looking at all of the property names of the first object in that array.
+
+It seems helpful to not only provide a similarly simplistic means of providing data to the `nimble-table`, but to also provide the type for the row data via a generic argument on the `Table` class itself, i.e.:
+
+```ts
+export class Table<TData> extends FoundationElement {
+    @observable
+    public data: TData[];
+}
+```
+
+This provides a couple of benefits:
+
+-   Interfacing with the TanStack APIs is now more direct and avoids `unknown` typing, e.g.:
+
+```ts
+public data: TData[] = [];
+private _options: TanstackOptionsResolved<TData>;
+
+this._options = {
+    get data(): TData[] { // instead of unknown[]
+        return data;
+    }
+}
+```
+
+-   View templating can access named columns in some scenarios. For instance, given a row type of `Person`, that has a field `friends` in it whose value was another table, a user could define a FAST ViewTemplate in the following way:
+
+```ts
+const rowTemplate = (index: number): ViewTemplate<any, Table<Person>> => html<
+    any,
+    Table<Person>
+>`
+    <nimble-table
+        style="max-height: 500px"
+        :data="${(_, c) => c.parent.tableData[index]!.row.original.friends}"
+    >
+    </nimble-table>
+`;
+```
+
+Both of these are demonstrated in the following prototype branch:
+
+[Table in a Table](https://github.com/ni/nimble/blob/tanstack-table-in-table) (see [Storybook](https://60e89457a987cf003efc0a5b-wzissaavew.chromatic.com/iframe.html?args=&id=table--table-story&viewMode=story))
+
+### Limitations
```

Comment:
I think it's implied but it would be good to explicitly mention that the API only allows setting data via a property, not an attribute.